ui/src/main/java/com/wireguard/android/activity/TvMainActivity.kt

```diff
@@ -317,10 +319,31 @@ class TvMainActivity : AppCompatActivity() {
         }
     }
 
+    suspend fun autoConnectToVpn() {
+        if (binding.tunnelList.size > 0) {
+            if (Application.getTunnelManager().lastUsedTunnel?.state != Tunnel.State.UP) {
+                Application.getTunnelManager().lastUsedTunnel?.let { tunnel ->
+                    setTunnelStateWithPermissionsResult(
+                        tunnel
+                    )
+                } ?: run {
+                    setTunnelStateWithPermissionsResult(
+                        Application.getTunnelManager().getTunnels().get(0)
+                    )
+                }
+            } else if (Application.getTunnelManager().lastUsedTunnel?.state == Tunnel.State.UP) {
+                openApp("com.example.nrg")
+            }
+        }
+    }
+
     private suspend fun updateStats() {
         binding.tunnelList.forEach { viewItem ->
+            if (binding.tunnelList.indexOfChild(viewItem) == 0) {
+                autoConnectToVpn()
+            }
             val listItem = DataBindingUtil.findBinding<TvTunnelListItemBinding>(viewItem)
-                    ?: return@forEach
+                ?: return@forEach
```

Comment:
Why did we modify this?

Reply:
It was changed above and somehow reflected change on this line